File: php/src/common/SoupBuffet.php
<?

#//SoupBuffet.java - A helper class

require_once 'src/common/require_all.php';

class SoupBuffet
{

	public $soupBuffetName;
	public $chickenSoup;
	public $clamChowder;
	public $fishChowder;
	public $minnestrone;
	public $pastaFazul;
	public $tofuSoup;
	public $vegetableSoup;

	function getTodaysSoups ()
	{
	  return " Today's Soups!  ".
		  " Chicken Soup: " .
		 $this->chickenSoup->soupName .
		  " Clam Chowder: " .
		 $this->clamChowder->soupName .
		  " Fish Chowder: " .
		 $this->fishChowder->soupName .
		  " Minnestrone: " .
		 $this->minnestrone->soupName .
		  " Pasta Fazul: " .
		 $this->pastaFazul->soupName .
		  " Tofu Soup: " .
		 $this->tofuSoup->soupName .
		  " Vegetable Soup: " .
		 $this->vegetableSoup->soupName;
	}

}

File: clojure/src/common/soupbuffet.clj
(ns common)

(defprotocol SoupOfTheDayBuffet (todays-soups [this]))
(defrecord soupbuffet [
	name 
	chickensoup
	clamchowder
	fishchowder
	minnestrone
	pastafazul
	tofusoup
	vegetablesoup]
	SoupOfTheDayBuffet(todays-soups [this]
	(str
		" Today's Soups!  "
		" Chicken Soup: " (:name (:chickensoup this))
      " Clam Chowder: " (:name (:clamchowder  this))
      " Fish Chowder: " (:name (:fishchowder  this))
      " Minnestrone: "  (:name (:minnestrone  this))
      " Pasta Fazul: "  (:name (:pastafazul  this))
      " Tofu Soup: "    (:name (:tofusoup  this))
      " Vegetable Soup: " (:name (:vegetablesoup this ))
	)))
